core/codenvy-hosted-api-workspace/src/main/java/com/codenvy/api/workspace/server/stack/StackPermissionsFilter.java

```diff
@@ -0,0 +1,82 @@
+/*
+ *  [2012] - [2016] Codenvy, S.A.
+ *  All Rights Reserved.
+ *
+ * NOTICE:  All information contained herein is, and remains
+ * the property of Codenvy S.A. and its suppliers,
+ * if any.  The intellectual and technical concepts contained
+ * herein are proprietary to Codenvy S.A.
+ * and its suppliers and may be covered by U.S. and Foreign Patents,
+ * patents in process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Codenvy S.A..
+ */
+package com.codenvy.api.workspace.server.stack;
+
+import org.eclipse.che.api.core.ForbiddenException;
+import org.eclipse.che.api.core.ServerException;
+import org.eclipse.che.api.workspace.server.stack.StackService;
+import org.eclipse.che.commons.env.EnvironmentContext;
+import org.eclipse.che.commons.user.User;
+import org.eclipse.che.everrest.CheMethodInvokerFilter;
+import org.everrest.core.Filter;
+import org.everrest.core.resource.GenericMethodResource;
+
+import javax.ws.rs.Path;
+
+/**
+ * Restricts access to methods of {@link StackService} by users' permissions
+ *
+ * <p>Filter should contain rules for protecting of all methods of {@link StackService}.<br>
+ * In case when requested method is unknown filter throws {@link ForbiddenException}
+ *
+ * @author Sergii Leschenko
+ */
+@Filter
+@Path("/stack{path:(/.*)?}")
+public class StackPermissionsFilter extends CheMethodInvokerFilter {
```

Comment:
I'm going to add them. Something like https://github.com/codenvy/codenvy/blob/master/core/codenvy-hosted-api-workspace/src/test/java/com/codenvy/api/workspace/server/filters/WorkspacePermissionsFilterTest.java#L70-70